src/app/_components/Header/Header.js

```diff
@@ -1,9 +1,29 @@
-import React from 'react';
+'use client';
+
+import React, { useState } from 'react';
+import Modal from '../Modal';
 
 export default function Header() {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const toggleModal = () => {
+    setIsModalOpen(prevState => !prevState);
+  };
   return (
-    <header className="bg-primary-100 p-5 text-h1 text-primary-700">
-      Header
+    <header className="flex justify-between bg-primary-100">
+      <h1 className="p-3 text-h1 text-primary-700"> Header </h1>
+      <button
+        onClick={() => toggleModal()}
```

Comment:
No reason to wrap callback
```suggestion
        onClick={toggleModal}
```